pkg/apis/metalkube/v1alpha1/baremetalhost_types.go

```diff
@@ -323,7 +324,17 @@ func (host *BareMetalHost) getLabel(name string) string {
 	return host.Labels[name]
 }
 
-// SetHardwareProfile updates the HardwareProfileLabel and returns
+// NeedsHardwareProfile returns true if the profile is not set
+func (host *BareMetalHost) NeedsHardwareProfile() bool {
```

Comment:
Can you just elaborate on why these two methods are valuable?

Reply:
I was trying not to encode the comparison to an empty string in the place where NeedsHardwareProfile() is called. It's trivial, but it's parallel to some of the other Needs*() methods.

Reply:
Gotcha. `""` is the zero value for a string in go, so while not explicitly an "unset" state, I think it's pretty close and regularly gets used as such. I think the code would be more idiomatic without these two methods, but that's subjective, and they're hardly showstoppers.